src/main/java/org/broadinstitute/barclay/argparser/CommandLinePluginDescriptor.java

```diff
@@ -135,6 +136,7 @@ public abstract Object getInstance(Class<?> pluggableClass)
      * @return true if the plugin for this class was specified on the command line, or the
      * values in this class may be set byt he user, otherwise false
      */
+    // TODO: I don't think that this should be bounded
     public abstract boolean isDependentArgumentAllowed(Class<?> dependentClass);
```

Comment:
I'm not sure what you mean by "implementation dependent" in this case. But I think the type bounds should stay like it is since this is used by the command line parser.